osu.Game/Rulesets/Objects/SliderPathExtensions.cs

```diff
@@ -95,8 +101,26 @@ private static void reverseControlPoints(this SliderPath sliderPath, out Vector2
                 else if (p.Type != null)
                     (p.Type, lastType) = (lastType, p.Type);
 
-                sliderPath.ControlPoints.Insert(0, p);
+                new InsertPathControlPointChange(sliderPath.ControlPoints, 0, p).Submit(changeHandler);
             }
         }
+
+        /// <summary>
+        /// Removes a range of <see cref="PathControlPoint"/>s from the provided <see cref="BindableList{T}"/>.
+        /// </summary>
+        public static void SubmitRemoveRange(this BindableList<PathControlPoint> controlPoints, int startIndex, int count, NewBeatmapEditorChangeHandler? changeHandler)
+        {
+            for (int i = 0; i < count; i++)
+                new RemovePathControlPointChange(controlPoints, startIndex).Submit(changeHandler);
+        }
+
+        /// <summary>
+        /// Adds a range of <see cref="PathControlPoint"/>s to the provided <see cref="BindableList{T}"/>.
+        /// </summary>
+        public static void SubmitAddRange(this BindableList<PathControlPoint> controlPoints, IEnumerable<PathControlPoint> points, NewBeatmapEditorChangeHandler? changeHandler)
+        {
+            foreach (var point in points)
+                new InsertPathControlPointChange(controlPoints, controlPoints.Count, point).Submit(changeHandler);
+        }
```

Comment:
Don't like these at all, they give me a faint waft of the proxy stuff. Just submit the commands directly.

Reply:
I see how it resembles the proxy stuff. I added these because it was easy to make mistakes in the implementation of these routines. I think I'll turn these methods into commands instead.